File: Backend/TruckEase/TruckEase/Contracts/CompanyUser/Mappers/CompanyUserMapper.cs
using TruckEase.Contracts.CompanyUser.Requests;
using TruckEase.Contracts.CompanyUser.Responses;
using TruckEase.Dtos;
using TruckEase.ValueObjects;
using TruckEase.Vtos;

namespace TruckEase.Contracts.CompanyUser.Mappers;

public static class CompanyUserMapper
{
    public static RegisterCompanyUserVto ToRegisterCompanyUserVto(this RegisterCompanyUserRequest request)
    {
        return new RegisterCompanyUserVto
        {
            Email = EmailValue.Create(request.Email),
            FirstName = FirstNameValue.Create(request.FirstName),
            LastName = LastNameValue.Create(request.LastName),
            Password = PasswordValue.Create(request.Password),
        };
    }

    public static EditCompanyUserVto ToEditCompanyUserVto(this EditCompanyUserRequest request)
    {
        return new EditCompanyUserVto
        {
            FirstName = FirstNameValue.Create(request.FirstName),
            LastName = LastNameValue.Create(request.LastName),
            WorkPhone = request.WorkPhone,
            MobilePhone = request.MobilePhone
        };
    }

    public static LoginCompanyUserVto ToLoginCompanyUserVto(this LoginRequest request)
    {
        return new LoginCompanyUserVto
        {
            Email = request.Email,
            Password = request.Password,
        };
    }

    public static RegisterCompanyUserWithTokenResponse ToRegisteredCompanyUserResponseWithToken(this RegisterCompanyUserWithTokenDto companyUserDto)
    {
        return new RegisterCompanyUserWithTokenResponse
        {
            FirstName = companyUserDto.FirstName,
            LastName = companyUserDto.LastName,
            Email = companyUserDto.Email,
            IsUserInCompany = companyUserDto.IsUserInCompany,
            Token = companyUserDto.Token
        };
    }

    public static RegisterCompanyUserResponse ToRegisteredCompanyUserResponse(this RegisterCompanyUserDto companyUserDto)
    {
        return new RegisterCompanyUserResponse
        {
            FirstName = companyUserDto.FirstName,
            LastName = companyUserDto.LastName,
            Email = companyUserDto.Email,
            IsUserInCompany = companyUserDto.IsUserInCompany,
        };
    }

    public static CompanyUserInfoResponse ToCompanyUserInfoResponse(this CompanyUserInfoDto companyUserDto)
    {
        return new CompanyUserInfoResponse(
            companyUserDto.FirstName,
            companyUserDto.LastName,
            companyUserDto.Email,
            companyUserDto.JobTitle,
            companyUserDto.WorkPhone,
            companyUserDto.MobilePhone);
    }

}

File: Frontend/TruckEase/TruckEaseApp/src/app/models/mappers/company-user.mapper.ts
import { CompanyUserWithTokenResponse } from "../responses/company-user-with-token.response";
import { RegisterCompanyUserResponse } from "../responses/company-user.response";
import { CompanyUserInfoViewModel } from "../view-models/company-user-info.view-model";
import { CompanyUserWithTokenViewModel } from "../view-models/company-user-with-token.view-model";
import { RegisterCompanyUserViewModel } from "../view-models/company-user.view-model";

export namespace CompanyUserMapper {

    export function toRegisterCompanyUserViewModel(registerCompanyUserResponse: RegisterCompanyUserResponse):
    RegisterCompanyUserViewModel {
  	return {
  		firstName: registerCompanyUserResponse.firstName,
  		lastName: registerCompanyUserResponse.lastName,
  		email: registerCompanyUserResponse.email,
  		isUserInCompany: registerCompanyUserResponse.isUserInCompany
  	};
  }

  export function toRegisterCompanyUserWithTokenViewModel(registerCompanyUserResponse: CompanyUserWithTokenResponse):
  CompanyUserWithTokenViewModel {
	return {
		firstName: registerCompanyUserResponse.firstName,
		lastName: registerCompanyUserResponse.lastName,
		email: registerCompanyUserResponse.email,
		isUserInCompany: registerCompanyUserResponse.isUserInCompany,
		token: registerCompanyUserResponse.token
	};
}

export function toCompanyUserInfoViewModel(companyUserInfo: CompanyUserInfoViewModel):
CompanyUserInfoViewModel {
  return {
	  firstName: companyUserInfo.firstName,
	  lastName: companyUserInfo.lastName,
	  email: companyUserInfo.email,
	  jobTitle: companyUserInfo.jobTitle,
	  workPhone: companyUserInfo.workPhone,
	  mobilePhone: companyUserInfo.mobilePhone
  };
}

}